internal/setup/setup.go

```diff
@@ -331,20 +339,20 @@ type yamlArchive struct {
 
 type yamlPackage struct {
 	Name    string               `yaml:"package"`
-	Archive string               `yaml:"archive"`
-	Slices  map[string]yamlSlice `yaml:"slices"`
+	Archive string               `yaml:"archive,omitempty"`
```

Comment:
Is there a specification for our YAML format? I lack the context to know why we are changing the fields to make them `omitempty`.

On the other hand if this is the decided upon format, we should have a test that fails if we accidentally remove the flag. When I tried removing it from `Archive`, the tests still passed.

Reply:
This is a good point. The spec for this feature does not include changes to the yaml format, so we might need to take this discussion offline.

Reply:
If the archive field is empty in the slice definition file, it is automatically set to the default (`ubuntu`). So it will never actually be empty while marshalling any package. And `omitempty` changes nothing while unmarshalling. I think that's why none of the tests failed.

About the addition of `omitempty` flags to all other fields, no there is not a specification about empty fields. Sure, if the slice definition file contains empty fields, parsing will work as intended with or without `omitempty`.

But what about marshalling the types to yaml? Do we print the empty fields or omit them? I do not think there is a clearly documented specification for that in this project yet. Usually, we do not include empty fields in the yaml. Hence, the addition of `omitempty`.